benchmark/src/main/java/com/amazon/opendistroforelasticsearch/sql/benchmark/utils/load/mysql/MysqlDataTransformer.java

```diff
@@ -32,6 +42,60 @@
    */
   @Override
   public DataFormat transformData(String dataPath) throws Exception {
-    return null;
+    final File path = new File(dataPath);
+    if (!path.exists() || !path.isDirectory()) {
+      throw new FileNotFoundException("Invalid Directory");
+    }
+
+    MysqlDataFormat result = new MysqlDataFormat();
+
+    // Create directory to store transformed csv files.
+    CommandExecution.executeCommand("mkdir " + dataPath + "mysql/");
+    final String transformedDataPath = dataPath + "mysql/";
+
+    for (String tableName : MysqlTpchSchema.schemaMap.keySet()) {
+      final File table = new File(dataPath + tableName + ".tbl");
+      if (!table.exists() || !table.isFile()) {
+        throw new FileNotFoundException(tableName + ".tbl not found");
+      }
+
+      final FileReader fileReader = new FileReader(table);
+      final BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+      int tableDataFilesIndex = 1;
+      String filename = tableName + "_data_" + tableDataFilesIndex++ + ".csv";
+
+      BufferedWriter writer = new BufferedWriter(
+          new FileWriter(transformedDataPath + filename, true));
+
+      long tableLineIndex = 1;
+
+      // Create list of csv files for all tables. Each file contains 1000 data rows.
+      try {
+        result.addFile(tableName, transformedDataPath + filename);
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+          List<String> argsList = Arrays.asList(line.split("\\|"));
+          String row = argsList.stream().map(s -> "\"" + s + "\"").collect(Collectors.joining(","));
```

Comment:
Do you need to use a Stream here, or can you just directly use a StringJoiner?

Reply:
removed stream